RATapi/models.py

```diff
@@ -51,36 +45,107 @@ def __str__(self):
         return table.get_string()
 
 
-class Background(RATModel):
-    """Defines the Backgrounds in RAT."""
+class Signal(RATModel):
+    """Base model for background & resolution signals."""
 
-    name: str = Field(default_factory=lambda: "New Background " + next(background_number), min_length=1)
-    type: TypeOptions = TypeOptions.Constant
+    name: str = Field(default="New Signal", min_length=1)
+    type: TypeOptions = Field(default=TypeOptions.Constant, frozen=True)
```

Comment:
I think it'd be a good idea to write a custom error message here - "field is frozen" isn't particularly helpful.